src/EFCore/Metadata/Internal/InternalTypeBaseBuilder.cs

```diff
@@ -551,6 +551,9 @@ public static bool IsCompatible(MemberInfo? newMemberInfo, PropertyBase existing
         {
             var property = properties[i];
             var typeConfigurationSource = property.GetTypeConfigurationSource();
+            var oldConfigurationSource = property.GetConfigurationSource();
```

Comment:
```suggestion
            var oldConfigurationSource = property.GetElementTypeConfigurationSource();
```

Though, in general this should've been handled by `Attach`

Reply:
This was left over from before I implemented Attach. It is not now needed.